lib/elixir/lib/record.ex

```diff
@@ -309,7 +317,9 @@ defmodule Record do
 
   # Callback invoked from record/0 and record/1 macros.
   @doc false
-  def __access__(atom, fields, args, caller) do
+  @spec __access__(tag, fields :: Keyword.t, args :: atom, caller :: Macro.Env) :: non_neg_integer
+  @spec __access__(tag, fields :: Keyword.t, args :: Keyword.t, caller :: Macro.Env) :: Macro.t
```

Comment:
@eksperimental I don't think it adds anything regardless of the multiple types. It's still the third argument, and that's called `args` in the signature anyways.